packages/artillery-plugin-publish-metrics/lib/newrelic/index.js

```diff
@@ -13,6 +13,18 @@ class NewRelicReporter {
       licenseKey: config.licenseKey
     };
 
+    if (!config.licenseKey) {
+      throw new Error(
```

Comment:
So I've actually been looking into this. 

Currently, if a user doesn't set a required config key (e.g. an API token), reporters behave in various ways. 
Some don't let the user know at all and let the test continue, others just log a message that can easily be missed while test is running, and then some let the test fail with error when attempting to send the data.

But I was thinking: if a crucial config setting is missing in the script, maybe the reporter should throw an error and stop the test run immediately. If a user sets the `publish-metrics` plugin with its `type` in a script I believe that shows clear intent to send the test data to one of the targets. It can be frustrating to see all your tests ran but no data arrived to destination, with no messages shown unless in debug mode. If we know the cause, we can fail early.

This is why I implemented this fix this way, and was actually going to ask if I can make this consistent throughout the reporters as it is a really easy thing to implement.

Let me know what you think about this, otherwise I can revert it. 🤓